src/DotVVM.Framework/Hosting/EmbeddedMarkupFileLoader.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Text;
+using DotVVM.Framework.Configuration;
+
+namespace DotVVM.Framework.Hosting
+{
+    public class EmbeddedMarkupFileLoader : IMarkupFileLoader
+    {
+        /// <summary>
+        /// Gets the markup file for the specified virtual path.
+        /// </summary>
+        public MarkupFile GetMarkup(DotvvmConfiguration configuration, string virtualPath)
+        {
+            if (virtualPath.IndexOf('.') == -1 || virtualPath.IndexOf('.') == 0)
+            {
+                return null;
+            }
+
+            string assemblyName = virtualPath.Substring(0, virtualPath.IndexOf("."));
```

Comment:
This won't work when the assembly name contains dots. You are looking for the first dot but actually you cannot know what is the name of the assembly. 
I think that the better strategy will be to look for all assemblies loaded in the current AppDomain and try to find the resource in all assemblies where the embedded resource name starts with the assembly name.
If you find multiple resources which satisfy the condition, an exception should be thrown.